compysition/actor.py

```diff
@@ -258,31 +295,33 @@ def __do_consume(self, function, event, queue):
                 event = new_event
 
             if self.REQUIRED_EVENT_ATTRIBUTES:
-                missing = [event.get(attribute) for attribute in self.REQUIRED_EVENT_ATTRIBUTES if not event.get(attribute, None)]
+                missing = [attribute for attribute in self.REQUIRED_EVENT_ATTRIBUTES if not event.get(attribute, None)]
                 if len(missing) > 0:
                     raise InvalidActorInput("Required incoming event attributes were missing: {missing}".format(missing=missing))
 
-            function(event, origin=queue.name, origin_queue=queue)
-        except QueueFull as err:
-            err.wait_until_free()
-            queue.put(event)
+            try:
+                function(event, origin=queue.name, origin_queue=queue)
+            except QueueFull as err:
+                err.queue.wait_until_free() # potential TypeError if target queue is not sent
+                queue.put(event) # puts event back into origin queue
         except InvalidActorInput as error:
             self.logger.error("Invalid input detected: {0}".format(error))
         except InvalidEventConversion:
             self.logger.error("Event was of type '{_type}', expected '{input}'".format(_type=type(event), input=self.input))
         except Exception as err:
             self.logger.warning("Event exception caught: {traceback}".format(traceback=traceback.format_exc()), event=event)
-            rescue_tracker = "{actor}_rescue_num".format(actor=self.name)
-            if self.rescue and event.get(rescue_tracker, 0) < self.max_rescue:
-                setattr(event, rescue_tracker, event.get(rescue_tracker, 0) + 1)
+            rescue_attribute = Actor._RESCUE_ATTRIBUTE_NAME_TEMPLATE.format(actor=self.name)
+            rescue_attempts =  event.get(rescue_attribute, 0)
+            if self.rescue and rescue_attempts < self.max_rescue:
+                setattr(event, rescue_attribute, rescue_attempts + 1)
                 sleep(1)
                 queue.put(event)
             else:
                 event.error = err
                 self.send_error(event)
 
     def create_event(self, *args, **kwargs):
-        if len(self.output) == 1:
+        if len(self.output) <= 1:
```

Comment:
What's this do? Seems like if `len(self.output) == 0` then we'll try to index into it and throw an unhandled `IndexError`. Is that handled somewhere that I'm not seeing?

Reply:
The motivation for this was the handling of `len(self.output) == 0` in the sense that it would raise the "too many output types" error.  Which didn't seem to depict the true nature of the problem.  After looking back through `Actor`, `len(self.output) == 0` would only happen if `output` was intentionally set to an empty tuple (`output = ( )`).  Regardless, I see your point in the sense that an `IndexError` might not be caught else where so I'm going to push a change real quick to address both sides.